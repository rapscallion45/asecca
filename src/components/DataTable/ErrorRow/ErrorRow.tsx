import { FC } from 'react';
import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import ErrorIcon from '@mui/icons-material/Error';
import { IDataTableColumn } from '@/components/DataTable/types';

/**
 * Data Table Error Row Props
 *
 * @since - 0.0.0
 *
 * @typedef IErrorRowProps
 * @prop {Array<IDataTableColumn>} columns - data table columns
 * @prop {string} message - error message string to display
 */
interface IErrorRowProps {
  columns: Array<IDataTableColumn>;
  message?: string;
}

/**
 * Data Table Error Row
 *
 * @author - [Carl Scrivener](https://github.com/rapscallion45)
 * @since - 0.0.0
 *
 * @param {IErrorRowProps} props - component props
 * @returns {FC} - data table error row functional component
 */
const ErrorRow: FC<IErrorRowProps> = (props) => {
  const { columns, message } = props;

  return (
    <TableRow>
      <TableCell colSpan={columns.length}>
        <Box display="flex" flexDirection="column" py={7} alignItems="center">
          <ErrorIcon color="error" fontSize="medium" />
          <Typography variant="body1" mt={2}>
            {message}
          </Typography>
        </Box>
      </TableCell>
    </TableRow>
  );
};

export default ErrorRow;
