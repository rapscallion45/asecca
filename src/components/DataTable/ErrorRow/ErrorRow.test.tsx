import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody } from '@mui/material';
import ErrorRow from './ErrorRow';
import dataTableColumns from '../../CostsConfigTable/costsConfigTableColumns';

/* default test error message */
const testMessage = 'This is an error message';

/**
 * Error Row Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 */
describe('Data Table Error Row', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <ErrorRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Message', () => {
    it('Should render the passed message text', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <ErrorRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByText(testMessage)).toBeInTheDocument();
    });
  });

  describe('Icon', () => {
    it('Should render error icon', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <ErrorRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByTestId('ErrorIcon')).toBeInTheDocument();
    });
  });
});
