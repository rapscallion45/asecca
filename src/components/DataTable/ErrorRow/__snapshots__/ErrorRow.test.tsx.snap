// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Error Row Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        colSpan={7}
      >
        <div
          className="MuiBox-root css-1lu6zdn"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
            data-testid="ErrorIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
          >
            This is an error message
          </p>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
